client/components/relatedWidget/Outfit.jsx

```diff
@@ -44,30 +36,20 @@ class Outfit extends React.Component {
     });
   }
 
-  onSelect(key) {
-    this.setState({ selected: key });
-  }
-
   buildCarousel() {
   this.productItems = new ProductList(
     {
       products: this.props.yourOutfit,
       images: this.state.image
     },
-      this.state.selected,
       this.props.getNewProduct);
   }
 
-  componentDidMount() {
-    this.getImage(this.props.product.id);
-    this.buildCarousel();
-  }
-
```

Comment:
Removed componentDidMount since it usually fires before it receives product.id which is undesired